public/app/plugins/datasource/loki/components/LokiQueryEditor.test.tsx

```diff
@@ -160,23 +186,31 @@ function renderWithMode(mode: QueryEditorMode) {
   return renderWithProps({ editorMode: mode });
 }
 
-function renderWithProps(overrides?: Partial<LokiQuery>) {
+function renderWithProps(overrides?: Partial<LokiQuery>, componentProps: Partial<LokiQueryEditorProps> = {}) {
   const query = defaultsDeep(overrides ?? {}, cloneDeep(defaultQuery));
   const onChange = jest.fn();
 
-  const stuff = render(<LokiQueryEditor {...defaultProps} query={query} onChange={onChange} />);
+  const allProps = { ...defaultProps, ...componentProps };
+  const stuff = render(<LokiQueryEditor {...allProps} query={query} onChange={onChange} />);
   return { onChange, ...stuff };
 }
 
 async function expectCodeEditor() {
-  // Label browser shows this until log labels are loaded.
-  expect(await screen.findByText('Loading...')).toBeInTheDocument();
+  expect(await screen.findByText('MonacoQueryFieldWrapper')).toBeInTheDocument();
 }
 
 async function expectBuilder() {
   expect(await screen.findByText('Label filters')).toBeInTheDocument();
 }
 
+function expectRunQueriesButton() {
+  expect(screen.queryByText('Run queries')).toBeInTheDocument();
+}
+
+function expectNoRunQueriesButton() {
+  expect(screen.queryByText('Run queries')).not.toBeInTheDocument();
+}
```

Comment:
let's use more semantic selectors:
```suggestion
function expectRunQueriesButton() {
  expect(screen.getByRole('button', { name: /run queries/i })).toBeInTheDocument();
}

function expectNoRunQueriesButton() {
  expect(screen.queryByRole('button', { name: /run queries/i })).not.toBeInTheDocument();
}
```